minitwit-backend/minitwit-backend.API/Data/UserRepository.cs

```diff
@@ -64,14 +64,18 @@ public async Task<int> VerifyLogin(UserLoginDTO userLoginDTO)
                 var hash = userFromDatabase.PwHash.Split(":")[1];
                 if (VerifyPassword(userLoginDTO.Password, hash, Convert.FromHexString(salt)))
                 {
-                    return userFromDatabase.UserId;
+                    return new UserDTO
+                    {
+                        UserId = userFromDatabase.UserId,
+                        UserName = userFromDatabase.Username
+                    };
                 }
                 else
                 {
-                    return -1;
+                    return null;
```

Comment:
You return a null object, but in the Controller you check if id = -1, that doesn't add up